content/tutorials/MQTT-snake-js.textile

```diff
@@ -0,0 +1,255 @@
+
+"See this step in Github":https://github.com/ably/tutorials/TODO
+
+h2. Step 7 - Recieving inputs from Ably
+
+Now that you've set up your controller, it's time to create something to run the game of Snake, and receive the data from Ably that you sent from the controller. You'll be using the normal Ably Realtime protocol for this part. This is possible due to the fact Ably is "Protocol Agnostic":https://support.ably.io/solution/articles/3000054126-do-you-support-mqtt-are-you-protocol-agnostic-, allowing for protocols to be used interchangebly.
+
+Firstly create a file called @snake.html@, and place the following code inside:
+
+```[html]
+  <!doctype html>
+  <html>
+    <head>
+      <title>Ably Realtime Snake</title>
+      <meta name="description" content="Fun Snake Game by Ably">
+      <meta name="keywords" content="ably realtime snake">
+      <link rel="stylesheet" href="game.css">
+      <script src="http://cdn.ably.io/lib/ably.min-1.js"></script>
+    </head>
+    <body>
+      <script src="snake.js"></script>
+      <h1 id="heading">Ably Realtime Snake</h1>
+      <canvas id="gc" height="400" width="400"></canvas>
+      </table>
+    </body>
+  </html>
+```
+
+Next, create another file called @snake.css@ containing the following:
+
+```[css]
+#heading {
+	color: #FFA500;
+	text-align: center;
+}
+
+#gc {
+    padding-left: 0;
+    padding-right: 0;
+    margin-left: auto;
+    margin-right: auto;
+    display: block;
+    width: "400";
+    height: "400";
+}
+```
+
+Now, create a file called @snake.js@, and put in the following code, replacing REPLACE_WITH_YOUR_API_KEY with your API Key:
+
+```[javascript]
+  var ably = new Ably.Realtime('REPLACE_WITH_YOUR_API_KEY');
+  var enc = new TextDecoder();
+  var channel = ably.channels.get('input');
+  channel.subscribe(function(message) {
+    var command = enc.decode(message.data);
+    console.log(command);
+  });
+```
+
+This will simply subscribe to the @input@ channel you're publishing to from the controller, and print it into the console. Open up @snake.html@ in a browser, open up the browser's "console":http://wickedlysmart.com/hfjsconsole/, and run your controller by typing @node controller.js@ into your command line. Now when you press the arrow keys from your controller, you should see messages being received by the browser in the console.
+
+TODO: Add image of controller input + console output
+
+"See this step in Github":https://github.com/ably/tutorials/TODO
+
+h2. Step 8 - Play Snake!
+
+With the setup complete, you can now add in the code for the actual snake game. Replace the contents of @snake.js@ with "this":https://github.com/ably/tutorials/TODO code, replacing REPLACE_WITH_YOUR_API_KEY with your API Key, and when you reload the @snake.html@ page you should see your basic Snake game. Run the controller as before, and you can now play! Press enter to start, arrow keys to move, and try to grow as large as you can.
```

Comment:
Perhaps remove 'and when you..' and put it as a new sentence like 'When you reload the ..'